.github/workflows/connector-test-command.yml

```diff
@@ -88,39 +88,60 @@ jobs:
         run: |
           # Get modified connectors
           CONNECTORS=$(./poe-tasks/get-modified-connectors.sh)
+          JAVA_CONNECTORS=$(./poe-tasks/get-modified-connectors.sh --java)
+          NON_JAVA_CONNECTORS=$(./poe-tasks/get-modified-connectors.sh --no-java)
           echo "Found connectors with changes:\n${CONNECTORS}"
           # Convert newlines → commas (and drop any trailing comma)
           CONNECTORS=$(printf '%s\n' "$CONNECTORS" \
             | tr '\n' ',' \
             | sed 's/,$//')
-          echo "Converted to comma-delimited list: ${CONNECTORS}"
+          JAVA_CONNECTORS=$(printf '%s\n' "$JAVA_CONNECTORS" \
+            | tr '\n' ',' \
+            | sed 's/,$//')
+          NON_JVM_CONNECTORS=$(printf '%s\n' "$NON_JAVA_CONNECTORS" \
+            | tr '\n' ',' \
+            | sed 's/,$//')
           echo CONNECTORS=${CONNECTORS} >> $GITHUB_ENV
+          echo JVM_CONNECTORS=${JVM_CONNECTORS} >> $GITHUB_ENV
+          echo NON_JVM_CONNECTORS=${NON_JVM_CONNECTORS} >> $GITHUB_ENV
 
-      - name: Convert to JSON
+      - name: Convert Connector Lists to JSON
         id: convert-matrix
         run: |
-          # Check if CONNECTORS is empty
-          if [ -z "$CONNECTORS" ]; then
-            echo "No connectors modified. Setting matrix to empty string."
-            echo "matrix=" >> $GITHUB_OUTPUT
-            exit 0
-          fi
-          # quote each comma‑separated connector and rebuild the list in one step
-          json_elements="\"${CONNECTORS//,/\",\"}\""
-          json_output="{ \"connector\": [${json_elements}] }"
-          echo "matrix=$json_output" | tee -a "$GITHUB_OUTPUT"
+          # helper: turn comma-delimited list into a JSON array
+          to_json() {
+            local list="$1"
+            if [ -z "$list" ]; then
+              echo "[]"
+            else
+              printf '%s\n' "$list" \
+                | tr ',' '\n' \
+                | jq -R . \
+                | jq -cs .
+            fi
+          }
+
+          connectors_matrix=$(to_json "$CONNECTORS")
+          jvm_connectors_matrix=$(to_json "$JVM_CONNECTORS")
+          non_java_connectors_matrix=$(to_json "$NON_JVM_CONNECTORS")
+
+          echo "connectors_matrix=$connectors_matrix" >> $GITHUB_OUTPUT
+          echo "jvm_connectors_matrix=$jvm_connectors_matrix" >> $GITHUB_OUTPUT
+          echo "non_jvm_connectors_matrix=$non_jvm_connectors_matrix" >> $GITHUB_OUTPUT
 
     outputs:
-      connector-matrix: ${{ steps.convert-matrix.outputs.matrix }}
+      connectors-matrix: ${{ steps.convert-matrix.outputs.connectors-matrix }}
+      jvm-connectors-matrix: ${{ steps.convert-matrix.outputs.jvm-connectors-matrix }}
+      non-jvm-connectors-matrix: ${{ steps.convert-matrix.outputs.non-jvm-connectors-matrix }}
 
-  connectors-test-matrix:
+  jvm-connectors-test-matrix:
     needs: [generate-matrix]
-    runs-on: ubuntu-24.04
+    runs-on: ubuntu-20.04-8core
```

Comment:
I have a theory we might be able to simplify by turning it into an action. Will noodle on it more tomorrow.

Reply:
I don't feel strongly inclined to refactor code into separate code files until the implementation itself proves stable. But yes, at least in the long run, refactoring into reusable actions should be helpful if there are enough implementations to make the extra call out to another action worthwhile.

For now, wdyt? Can we iterate here in one place until stable?